### ISSUE TITLE
Remove legacy error notifications infrastructure in favor of store-based one

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3406]
The `storeErrorNotifications` feature flag should be removed and the logic behind it should become the new default, more specifically the only version available.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### Part 1: Replace generic errors legacy infrastructure

1. The principal usage of `showErrorNotification` should be replaced with a datastore-driven alternative, using the `core/site` store:
    1. It should receive actions `setInternalServerError( error )` and `clearInternalServerError()` and a selector `getInternalServerError()`, all of which behave similarly to their `*AuthError` counterparts in the `core/user` store.
    2. The usages of `showErrorNotification` that handle errors after a failed module activation (which is basically almost all usages of that function in total) should use `setInternalServerError` instead (also see next bullet point for context on how to replace the notifications themselves).
    3. A new `InternalServerError` component should be implemented in `assets/js/components/notifications/` which should display the error notification for `getInternalServerError` (inspired by logic in the `AuthError` component), with similar appearance (e.g. title, format etc.) that the notifications from the current usages of `showErrorNotification` have. That component should then be included in the `ErrorNotifications` component, on top of the two other components.
2. The `showErrorNotification` function and its remaining usages should be completely removed.
    1. The now unused `GenericError` component should also be removed.

### Part 2: Remove legacy error notifications infrastructure

1. The `storeErrorNotifications` feature flag should be removed and what's currently behind it should become the new way to display error notifications.
2. The legacy `ErrorNotification` component (sometimes referred to as `LegacyErrorNotifications`) should no longer be used.
3. The `googlesitekit.ErrorNotification` filter and all of its usages should be removed, e.g.:
    1. the legacy `ErrorNotification`
    2. the relevant logic in the legacy `handleWPError` function and in `assets/js/components/legacy-notifications/index.js`

## Implementation Brief

Numbering here matches numbering on AC. Refer to AC first

#### Part 1

1 .i. Create a new datastore `assets/js/googlesitekit/data/site/errors.js`.  Copy relevant actions and selectors from `assets/js/googlesitekit/datastore/user/authentication.js` but change names and constants (and ignore all the code for storing authentication state. This datastore is much larger)

1 .ii. Follow AC. Some existing calls to `showErrorNotification` have a second argument e.g.
```
showErrorNotification( GenericError, {
	id: 'activate-module-error',
	title: __( 'Internal Server Error', 'google-site-kit' ),
	description: error.message,
	format: 'small',
	type: 'win-error',
} );
```

and another call passes in an instantiated Notification component `showErrorNotification( InvalidCredentialsWarning );` 

The `id`, `title`, `description`, `learnMoreURL`, and `learnMoreLabel` change

So copying `AuthError` we'd want these passed in `error.data` 

1. iii. Follow IB. Copy like `AuthError` but pass in the keys above to the inner `Notification
component.

2. i. Follow AC

_"The usages of showErrorNotification that handle errors after a failed module activation (which is basically almost all usages of that function in total)"_
The only other usage of `showErrorNotification` in `assets/js/components/legacy-notifications/data-error.js` is to be deleted

#### Part 2
1. Remove `storeErrorNotifications` from `feature-flags.json`

2. Remove any usage of this `storeErrorNotifications` flag from codebase. Not many uses. In `Header.js` make the component behind the flag always visible.  Tests should still pass where previously this was set.

3. Remove <LegacyErrorNotification/> from `Header.js` (double-check no more instances)

4. Do as per AC

### Test Coverage

* Create a `assets/js/googlesitekit/data/site/errors.test.js`, copying relevant tests from `assets/js/googlesitekit/data/site/authentication.test.js` 
* Currently `assets/js/util/test/showErrorNotification.js` contains tests for `showErrorNotification`. Check that `setInternalServerError`, `clearInternalServerError` and `getInternalServerError` cause corresponding changes in `InternalServerError` (displayed, not displayed, displays correct message)

### Visual Regression Changes

* Changes not expected

## QA Brief

* Check the new `datastore/site/errors` store has been created with the following:
  * action: `setInternalServerError`
  * action: `clearInternalServerError`
  * selector:  `getInternalServerError`
* Check that all instances of `showErrorNotification` have been replaced with dispatches to `setInternalServerError`
* Check the  `storeErrorNotifications` flag has been removed from `feature-flags.json` and from everywhere in the codebase
* Check the `LegacyErrorNotification` component has been removed from `Header.js` and everywhere else in the codebase
* Check that js tests pass

## Changelog entry

* N/A

**Comment by danielgent** [comment]
@felixarntz  _"The usages of showErrorNotification that handle errors after a failed module activation (which is basically almost all usages of that function in total)"_

The only one that I see that isn't for a failed module activation is in `assets/js/components/legacy-notifications/data-error.js`

```
if ( 'Invalid Credentials' === errorMessage ) {
	showErrorNotification( InvalidCredentialsWarning );
}
```

What do we do with this? Just delete it? :thinking:

**Comment by felixarntz** [comment]
@danielgent IB mostly lgtm. Just a few things to clarify:
> Create a new datastore `assets/js/googlesitekit/data/site/error-notifications.js`

Nitpick, but maybe name this file a simpler `errors.js` - it stores errors, not necessarily notifications (even if we display such an error in a notification). When changing this, the name of the respective test file of course needs to change too.

> The only one that I see that isn't for a failed module activation is in `assets/js/components/legacy-notifications/data-error.js`
>
> ```
> if ( 'Invalid Credentials' === errorMessage ) {
>	showErrorNotification( InvalidCredentialsWarning );
> }
> ```
>
> What do we do with this? Just delete it? :thinking: 

Yes, indeed this can just be removed as it's no longer relevant - it's part of the legacy data API which will soon be removed in its entirety anyway.

> Remove `storeErrorNotifications` flag use throughout codebase. Not many uses. In `Header.js` make the component behind the flag always visible.  Tests should still pass where previously this was set.

Just to clarify, this also includes removing the use of `LegacyErrorNotification` in `Header.js` - maybe you were meaning that, but that is the old way of displaying notifications, which is now being replaced, so it needs to be removed.

Regarding the feature flag, keep in mind it also needs to be removed in the `feature-flags.json` file, not just the JS code - again, just to clarify.

**Comment by danielgent** [comment]
If anyone wants to pick this IB up, here is the PR WIP where I've been working on it https://github.com/google/site-kit-wp/pull/3425

**Comment by danielgent** [comment]
*_name this file a simpler errors.js_  :heavy_check_mark: 
*_include removing the use of LegacyErrorNotification_ :heavy_check_mark:

**Comment by felixarntz** [comment]
@danielgent The IB now has `error.js` - let's use `errors.js` (maybe just a typo?) :) Also please update the name of the test file for that file in the IB accordingly, that is currently still using `error-notifications.test.js`. After those updates, this is good to go! 👍

**Comment by felixarntz** [comment]
IB ✅

**Comment by Hazlitte** [comment]
@danielgent @felixarntz I just wanted to confirm that I should be creating the new `errors.js` store in a new `googlesitekit/data/site` folder and not in the existing `googlesitekit/datastore/site` one.

**Comment by danielgent** [comment]
@Hazlitte  that's definitely a typo on my behalf :smile:   Apologies. Put any more questions in here!

**Comment by Hazlitte** [comment]
Hi @danielgent.  During code review @eugene-manuilov pointed out that I can't remove this filter call to `googlesitekit.ErrorNotification`:

https://github.com/google/site-kit-wp/blob/8e9e890ad996bf7387007df2f9061a8e9f42827e/assets/js/components/legacy-notifications/index.js#L49-L53

I need to replicate the functionality that exists with the new `getInternalServerError` functionality.  From the IB I can't figure out what I should do in this instance.  Can you advise?

**Comment by Hazlitte** [comment]
Hi @felixarntz.
I have been executing this ticket and there is an aspect of it which I am unsure how to approach.  I have reviewed the AC and IB and it isn't clear to me how I should go about replacing calls to the `googlesitekit.ErrorNotification` filter which are in functions.  As these are not in exported components I am unable to make `useDispatch` calls to the store.  Here is an example:
https://github.com/google/site-kit-wp/blob/e2564bf10f06d36b04c36ece7f66e2edbaf78f3f/assets/js/components/data/index.js#L199-L204

**Comment by Hazlitte** [comment]
@eugene-manuilov @aaemnnosttv As this ticket is tagged as Sprint 52 and I am not working on Site Kit tomorrow would you like me to unassign myself?  I have updated the tagged branch with TODOs where there are aspects that still need addressing.

**Comment by felixarntz** [comment]
@eugene-manuilov @aaemnnosttv Please do not merge this until `develop` is open towards the 1.38.0 release, since we want to time this to go out together with #2258.

**Comment by danielgent** [comment]
## QA :white_check_mark:

Have checked all bullet points and all looks good to me
